nixos/modules/tasks/network-interfaces.nix

```diff
@@ -139,6 +139,16 @@ let
         description = "The default gateway metric/preference.";
       };
 
+      srcAddress= mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "1.1.1.1";
+        description = ''
+          If the interface has several ip addresses we can select by this option
```

Comment:
The description is a bit complicated, I would rename this as `The source address for outgoing connection`